internal/baggageutil/context.go

```diff
@@ -0,0 +1,37 @@
+package baggageutil
+
+import (
+	"context"
+
+	"github.com/quay/zlog"
+	"go.opentelemetry.io/otel/baggage"
+)
+
+// ContextWithValues adds a list of key value pairs to the context's baggage. kvPairs must be of even length.
+func ContextWithValues(ctx context.Context, kvPairs ...string) context.Context {
+	if len(kvPairs) == 0 {
+		return ctx
+	}
+
+	b := baggage.FromContext(ctx)
+	oldMembers := b.Members()
+	newMembers := make([]baggage.Member, 0, len(oldMembers)+len(kvPairs)/2)
+	newMembers = append(newMembers, oldMembers...)
```

Comment:
What's the reasoning for doing this? Preventing a bunch of copies?

If this is used everywhere, it's probably worth being hoisted up into zlog itself.

Reply:
Yeah, I wanted to avoid repeated `SetMember` calls, as that copies all existing members each time, leading to quadratic complexity.